improver/precipitation_type/shower_condition_probability.py

```diff
@@ -0,0 +1,182 @@
+
+    def _extract_inputs(self, cubes):
+        """Extract the required input cubes from the input cubelist and check
+        they are as required."""
+        try:
+            (cloud,) = cubes.extract(self.cloud_constraint)
+            (convection,) = cubes.extract(self.convection_constraint)
+        except ValueError:
+            input_cubes = ", ".join([cube.name() for cube in cubes])
+            msg = (
+                "A cloud area fraction and convective ratio are required, "
+                f"but the inputs were: {input_cubes}"
+            )
+            raise ValueError(msg)
+
+        if cloud.shape != convection.shape:
+            msg = (
+                "The cloud are fraction and convective ratio cubes are not "
+                "the same shape and cannot be combined to generate a shower"
+                " probability"
+            )
+            raise ValueError(msg)
+        return cloud, convection
+
+    def process(self, cubes: CubeList) -> Cube:
+        """
+        Create a shower condition probability from cloud fraction and convective
+        ratio fields. This plugin thresholds the two input diagnostics,
+        creates a hybrid probability field from the resulting binary fields,
+        and then collapses the realizations to give a non-binary probability
+        field that represents the likelihood of conditions being showery.
+
+        Args:
+            cubes:
+                A cubelist containing a cube of cloud fraction and one of
+                convective ratio.
+
+        Returns:
+            Probability of any precipitation being classified as showery
+
+        Raises:
+            ValueError: If inputs are not those expected.
+            ValueError: If the input cubes have different shapes, perhaps due
+                        to a missing realization in one and not the other.
+        """
+        cloud, convection = self._extract_inputs(cubes)
+
+        # Threshold cubes
+        cloud_thresholded = BasicThreshold(
+            self.cloud_threshold, comparison_operator="<="
+        ).process(cloud)
+        convection_thresholded = BasicThreshold(self.convection_threshold).process(
+            convection
+        )
+
+        # Fill any missing data in the convective ratio field with zeroes.
+        if np.ma.is_masked(convection_thresholded.data):
+            convection_thresholded.data = convection_thresholded.data.filled(0)
+
+        # Create a combined field taking the maximum of each input
+        shower_probability = np.maximum(
+            cloud_thresholded.data, convection_thresholded.data
+        ).astype(FLOAT_DTYPE)
+
+        # Create a new diagnostic cube containing the new data
+        template, attributes = self._output_metadata(convection_thresholded)
+        result = create_new_diagnostic_cube(
+            template.name(),
+            "1",
+            template,
+            mandatory_attributes=attributes,
+            data=shower_probability,
+        )
```

Comment:
I think I would prefer to put all this into the `_output_metadata` method (and rename it), but I'm not that bothered if you feel it should stay separate.